### ISSUE TITLE
This implementation is not part of the Windows Platform FIPS validated cryptographic algorithms.

### ISSUE THREAD

**Issue by gabeworden** [MscrmTools/XrmToolBox#403]
We are receiving the following error.  

See the end of this message for details on invoking 
just-in-time (JIT) debugging instead of this dialog box.

*************\* Exception Text **************
System.ComponentModel.Composition.CompositionException: The composition produced a single composition error. The root cause is provided below. Review the CompositionException.Errors property for more detailed information.

1) This implementation is not part of the Windows Platform FIPS validated cryptographic algorithms.

Resulting in: Exception has been thrown by the target of an invocation.

Resulting in: The type initializer for 'LibZ.Injected.AsmZResolver' threw an exception.

Resulting in: The type initializer for '<Module>' threw an exception.

Resulting in: An exception occurred while trying to create an instance of type 'MsCrmTools.ChartManager.Plugin'.

Resulting in: Cannot activate part 'MsCrmTools.ChartManager.Plugin'.
Element: MsCrmTools.ChartManager.Plugin -->  MsCrmTools.ChartManager.Plugin -->  DirectoryCatalog (Path="C:\Temp\XrmToolBox\Plugins")

Resulting in: Cannot get export 'MsCrmTools.ChartManager.Plugin (ContractName="XrmToolBox.Extensibility.Interfaces.IXrmToolBoxPlugin")' from part 'MsCrmTools.ChartManager.Plugin'.
Element: MsCrmTools.ChartManager.Plugin (ContractName="XrmToolBox.Extensibility.Interfaces.IXrmToolBoxPlugin") -->  MsCrmTools.ChartManager.Plugin -->  DirectoryCatalog (Path="C:\Temp\XrmToolBox\Plugins")

   at System.ComponentModel.Composition.Hosting.CompositionServices.GetExportedValueFromComposedPart(ImportEngine engine, ComposablePart part, ExportDefinition definition)
   at System.ComponentModel.Composition.Hosting.CatalogExportProvider.GetExportedValue(CatalogPart part, ExportDefinition export, Boolean isSharedPart)
   at System.ComponentModel.Composition.Primitives.Export.get_Value()
   at System.ComponentModel.Composition.ExportServices.GetCastedExportedValue[T](Export export)
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.LazyInitValue()
   at XrmToolBox.MainForm.CreateModel[T](Lazy`2 plugin, Int32& top, Int32 width, Int32 count)
   at XrmToolBox.MainForm.DisplayPlugins(Object filter)
   at XrmToolBox.MainForm.<MainForm_Load>d__30.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.<>c.<ThrowAsync>b__6_0(Object state)

*************\* Loaded Assemblies **************
mscorlib
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1063.1 built by: NETFXREL3STAGE
##     CodeBase: file:///C:/Windows/Microsoft.NET/Framework64/v4.0.30319/mscorlib.dll

XrmToolBox
    Assembly Version: 1.2016.3.30
    Win32 Version: 1.2016.3.30
##     CodeBase: file:///C:/Temp/XrmToolBox/XrmToolBox.exe

System.Windows.Forms
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0 built by: NETFXREL2
##     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll

System
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0 built by: NETFXREL2
##     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll

System.Drawing
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0 built by: NETFXREL2
##     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll

XrmToolBox.Extensibility
    Assembly Version: 1.2016.3.30
    Win32 Version: 1.2016.3.30
##     CodeBase: file:///C:/Temp/XrmToolBox/XrmToolBox.Extensibility.DLL

Microsoft.IdentityModel
    Assembly Version: 3.5.0.0
    Win32 Version: 10.0.10586.0
##     CodeBase: file:///C:/WINDOWS/assembly/GAC_MSIL/Microsoft.IdentityModel/3.5.0.0__31bf3856ad364e35/Microsoft.IdentityModel.dll

System.Core
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0 built by: NETFXREL2
##     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll

System.Configuration
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0 built by: NETFXREL2
##     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll

System.Xml
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0 built by: NETFXREL2
##     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll

McTools.Xrm.Connection
    Assembly Version: 1.2016.3.29
    Win32 Version: 1.2016.3.29
##     CodeBase: file:///C:/Temp/XrmToolBox/McTools.Xrm.Connection.DLL

McTools.Xrm.Connection.WinForms
    Assembly Version: 1.2016.3.29
    Win32 Version: 1.2016.3.29
##     CodeBase: file:///C:/Temp/XrmToolBox/McTools.Xrm.Connection.WinForms.DLL

Microsoft.Xrm.Sdk
    Assembly Version: 8.0.0.0
    Win32 Version: 8.0.1.39
##     CodeBase: file:///C:/Temp/XrmToolBox/Microsoft.Xrm.Sdk.DLL

System.ComponentModel.Composition
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0
##     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.ComponentModel.Composition/v4.0_4.0.0.0__b77a5c561934e089/System.ComponentModel.Composition.dll

DamSim.SolutionTransferTool
    Assembly Version: 1.2016.3.30
    Win32 Version: 1.2016.3.30
##     CodeBase: file:///C:/TEMP/XRMTOOLBOX/PLUGINS/DAMSIM.SOLUTIONTRANSFERTOOL.DLL

DamSim.ViewTransferTool
    Assembly Version: 1.2016.3.30
    Win32 Version: 1.2016.3.30
##     CodeBase: file:///C:/TEMP/XRMTOOLBOX/PLUGINS/DAMSIM.VIEWTRANSFERTOOL.DLL

GapConsulting.PowerBIOptionSetAssistant
    Assembly Version: 1.2016.3.30
    Win32 Version: 1.2016.3.30
##     CodeBase: file:///C:/TEMP/XRMTOOLBOX/PLUGINS/GAPCONSULTING.POWERBIOPTIONSETASSISTANT.DLL

Javista.XrmToolBox.ImportNN
    Assembly Version: 1.2016.3.30
    Win32 Version: 1.2016.3.30
##     CodeBase: file:///C:/TEMP/XRMTOOLBOX/PLUGINS/JAVISTA.XRMTOOLBOX.IMPORTNN.DLL

MsCrmTools.AccessChecker
    Assembly Version: 1.2016.3.30
    Win32 Version: 1.2016.3.30
##     CodeBase: file:///C:/TEMP/XRMTOOLBOX/PLUGINS/MSCRMTOOLS.ACCESSCHECKER.DLL

MsCrmTools.AssemblyRecoveryTool
    Assembly Version: 1.2016.3.30
    Win32 Version: 1.2016.3.30
##     CodeBase: file:///C:/TEMP/XRMTOOLBOX/PLUGINS/MSCRMTOOLS.ASSEMBLYRECOVERYTOOL.DLL

MsCrmTools.AttributeBulkUpdater
    Assembly Version: 1.2016.3.30
    Win32 Version: 1.2016.3.30
##     CodeBase: file:///C:/TEMP/XRMTOOLBOX/PLUGINS/MSCRMTOOLS.ATTRIBUTEBULKUPDATER.DLL

MsCrmTools.AuditCenter
    Assembly Version: 1.2016.3.30
    Win32 Version: 1.2016.3.30
##     CodeBase: file:///C:/TEMP/XRMTOOLBOX/PLUGINS/MSCRMTOOLS.AUDITCENTER.DLL

MsCrmTools.ChartManager
    Assembly Version: 1.2016.3.30
    Win32 Version: 1.2016.3.30
##     CodeBase: file:///C:/TEMP/XRMTOOLBOX/PLUGINS/MSCRMTOOLS.CHARTMANAGER.DLL

MsCrmTools.FetchXmlTester
    Assembly Version: 1.2016.3.30
    Win32 Version: 1.2016.3.30
##     CodeBase: file:///C:/TEMP/XRMTOOLBOX/PLUGINS/MSCRMTOOLS.FETCHXMLTESTER.DLL

MsCrmTools.FlsBulkUpdater
    Assembly Version: 1.2016.3.30
    Win32 Version: 1.2016.3.30
##     CodeBase: file:///C:/TEMP/XRMTOOLBOX/PLUGINS/MSCRMTOOLS.FLSBULKUPDATER.DLL

MsCrmTools.FormRelated
    Assembly Version: 1.2016.3.30
    Win32 Version: 1.2016.3.30
##     CodeBase: file:///C:/TEMP/XRMTOOLBOX/PLUGINS/MSCRMTOOLS.FORMRELATED.DLL

MsCrmTools.Iconator
    Assembly Version: 1.2016.3.30
    Win32 Version: 1.2016.3.30
##     CodeBase: file:///C:/TEMP/XRMTOOLBOX/PLUGINS/MSCRMTOOLS.ICONATOR.DLL

MsCrmTools.MetadataBrowser
    Assembly Version: 1.2016.3.30
    Win32 Version: 1.2016.3.30
##     CodeBase: file:///C:/TEMP/XRMTOOLBOX/PLUGINS/MSCRMTOOLS.METADATABROWSER.DLL

MsCrmTools.MetadataDocumentGenerator
    Assembly Version: 1.2016.3.30
    Win32 Version: 1.2016.3.30
##     CodeBase: file:///C:/TEMP/XRMTOOLBOX/PLUGINS/MSCRMTOOLS.METADATADOCUMENTGENERATOR.DLL

MsCrmTools.SampleTool
    Assembly Version: 1.2016.3.30
    Win32 Version: 1.2016.3.30
##     CodeBase: file:///C:/TEMP/XRMTOOLBOX/PLUGINS/MSCRMTOOLS.SAMPLETOOL.DLL

MsCrmTools.ScriptsFinder
    Assembly Version: 1.2016.3.30
    Win32 Version: 1.2016.3.30
##     CodeBase: file:///C:/TEMP/XRMTOOLBOX/PLUGINS/MSCRMTOOLS.SCRIPTSFINDER.DLL

MsCrmTools.SecurityRelated
    Assembly Version: 1.2016.3.30
    Win32 Version: 1.2016.3.30
##     CodeBase: file:///C:/TEMP/XRMTOOLBOX/PLUGINS/MSCRMTOOLS.SECURITYRELATED.DLL

MsCrmTools.SiteMapEditor
    Assembly Version: 1.2016.3.30
    Win32 Version: 1.2016.3.30
##     CodeBase: file:///C:/TEMP/XRMTOOLBOX/PLUGINS/MSCRMTOOLS.SITEMAPEDITOR.DLL

MsCrmTools.SolutionComponentsMover
    Assembly Version: 1.2016.3.30
    Win32 Version: 1.2016.3.30
##     CodeBase: file:///C:/TEMP/XRMTOOLBOX/PLUGINS/MSCRMTOOLS.SOLUTIONCOMPONENTSMOVER.DLL

MsCrmTools.SolutionImport
    Assembly Version: 1.2016.3.30
    Win32 Version: 1.2016.3.30
##     CodeBase: file:///C:/TEMP/XRMTOOLBOX/PLUGINS/MSCRMTOOLS.SOLUTIONIMPORT.DLL

MscrmTools.SyncFilterManager
    Assembly Version: 1.2016.3.30
    Win32 Version: 1.2016.3.30
##     CodeBase: file:///C:/TEMP/XRMTOOLBOX/PLUGINS/MSCRMTOOLS.SYNCFILTERMANAGER.DLL

MsCrmTools.SynchronousEventOrderEditor
    Assembly Version: 1.2016.3.30
    Win32 Version: 1.2016.3.30
##     CodeBase: file:///C:/TEMP/XRMTOOLBOX/PLUGINS/MSCRMTOOLS.SYNCHRONOUSEVENTORDEREDITOR.DLL

MsCrmTools.Translator
    Assembly Version: 1.2016.3.30
    Win32 Version: 1.2016.3.30
##     CodeBase: file:///C:/TEMP/XRMTOOLBOX/PLUGINS/MSCRMTOOLS.TRANSLATOR.DLL

MsCrmTools.UserSettingsUtility
    Assembly Version: 1.2016.3.30
    Win32 Version: 1.2016.3.30
##     CodeBase: file:///C:/TEMP/XRMTOOLBOX/PLUGINS/MSCRMTOOLS.USERSETTINGSUTILITY.DLL

MsCrmTools.ViewLayoutReplicator
    Assembly Version: 1.2016.3.30
    Win32 Version: 1.2016.3.30
##     CodeBase: file:///C:/TEMP/XRMTOOLBOX/PLUGINS/MSCRMTOOLS.VIEWLAYOUTREPLICATOR.DLL

MsCrmTools.WebResourcesManager
    Assembly Version: 1.2016.3.30
    Win32 Version: 1.2016.3.30
##     CodeBase: file:///C:/TEMP/XRMTOOLBOX/PLUGINS/MSCRMTOOLS.WEBRESOURCESMANAGER.DLL

System.Design
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0 built by: NETFXREL2
##     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Design/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Design.dll

System.Runtime.Serialization
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0 built by: NETFXREL2
##     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Runtime.Serialization/v4.0_4.0.0.0__b77a5c561934e089/System.Runtime.Serialization.dll

Microsoft.Crm.Sdk.Proxy
    Assembly Version: 8.0.0.0
    Win32 Version: 8.0.0001.0079
##     CodeBase: file:///C:/Temp/XrmToolBox/Microsoft.Crm.Sdk.Proxy.DLL

MetadataViewProxies_cadec60a-25a5-4668-afd5-3fb7c5d71d0d
    Assembly Version: 0.0.0.0
    Win32 Version: 4.6.1038.0
##     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.ComponentModel.Composition/v4.0_4.0.0.0__b77a5c561934e089/System.ComponentModel.Composition.dll

*************\* JIT Debugging **************
To enable just-in-time (JIT) debugging, the .config file for this
application or computer (machine.config) must have the
jitDebugging value set in the system.windows.forms section.
The application must also be compiled with debugging
enabled.

For example:

<configuration>
    <system.windows.forms jitDebugging="true" />
</configuration>

When JIT debugging is enabled, any unhandled exception
will be sent to the JIT debugger registered on the computer
rather than be handled by this dialog box.

**Comment by MscrmTools** [comment]
Never got this error before... Does the same error occur for other plugins?

**Comment by gabeworden** [comment]
I received this just in the plugin that are included when I downloaded the latest toolbox.  
On Apr 25, 2016 1:16 PM, MscrmTools <notifications@github.com> wrote:Never got this error before... Does the same error occur for other plugins?

**Comment by MscrmTools** [comment]
Did you unblock files before using them?

**Comment by gabeworden** [comment]
yes I  did.  I deleted a couple of plugins from XrmToolbox that were listed in the errors, and now it loads. I don’t know if those missing assemblies will cause it to fail eventually.  , I removed the following 2 plugins:
1) MsCrmTools.ChartManager.dll
2) MsCrmTools.WebResourcesManager.dll

But although the XrmToolbox now load, when I add the FetchXMLBuilder plugin assemblies, those do not load, and that is an important capability.

**Comment by rappen** [comment]
@gabeworden regarding FetchXML Builder, have you tried the trick described here? https://github.com/Cinteros/FetchXMLBuilder/issues/120

**Comment by gabeworden** [comment]
I downloaded XrmToolBox.prerelease_v1.2016.0427.143 and the issue is still present.  The following the files are not FIPS 104-2 compliant; MsCrmTools.ChartManager.dll and MsCrmTools.WebResourcesManager.dll

**Comment by MscrmTools** [comment]
It seems the FIPS policy is something handled by your IT
And it seems it occurs with assemblies merged with LibZ
Not something I can debug easily since I'm not in this configuration

Here is a workaround : http://stackoverflow.com/questions/14509354/this-implementation-is-not-part-of-the-windows-platform-fips-validated-cryptogra

**Comment by MscrmTools** [comment]
As I did not get any feedback since more than a year, I'm closing this issue
